src/arrayMethodReduce.js

```diff
@@ -1,11 +1,28 @@
 'use strict';
 
-/**
- * Implement method Reduce
- */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let acc;
+
+    if (initialValue !== undefined) {
+      acc = initialValue;
+    } else if (arguments.length > 1) {
+      acc = initialValue;
+    } else {
+      acc = this[0];
+    }
+
+    if (acc === this[0]) {
+      for (let i = 1; i < this.length; i++) {
+        acc = callback(acc, this[i], i, this);
+      }
+    } else {
+      for (let i = 0; i < this.length; i++) {
+        acc = callback(acc, this[i], i, this);
```

Comment:
Leave only this cycle, the other one has been issued. The condition is not needed because you changed `acc = this[0]` to `acc = 0`

Reply:
Accepted)

Reply:
But you have similar code, so remove redundant repeats and change conditionally  only index value 
![image](https://github.com/mate-academy/js_array-method-reduce/assets/70009205/53663034-769f-458e-aa75-f8c5f7d69c67)